lib/fulfil/response_handler.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Fulfil
+  # The `Fulfil::ResponseHandler` is parses the HTTP response from Fulfil. If it
+  # encounters an HTTP status code that indicates an error, it will raise an internal
+  # exception that the consumer can catch.
+  #
+  # @example
+  #   Fulfil::ResponseHandler.new(@response).verify!
+  #   => { id: 100 }
+  #
+  #   Fulfil::ResponseHandler.new(@response).verify!
+  #   => Fulfil::Error::BadRequest
+  class ResponseHandler
+    HTTP_ERROR_CODES = {
+      400 => Fulfil::HttpError::BadRequest,
+      401 => Fulfil::HttpError::AuthorizationRequired,
+      402 => Fulfil::HttpError::PaymentRequired,
+      403 => Fulfil::HttpError::Forbidden,
+      404 => Fulfil::HttpError::NotFound,
+      405 => Fulfil::HttpError::MethodNotAllowed,
+      406 => Fulfil::HttpError::NotAccepted,
+      422 => Fulfil::HttpError::UnprocessableEntity,
+      429 => Fulfil::HttpError::TooManyRequests,
+      500 => Fulfil::HttpError::InternalServerError
+    }.freeze
+
+    def initialize(response)
+      @response = response
+      @status_code = response.code
+    end
+
+    def verify!
+      return true unless HTTP_ERROR_CODES.key?(@status_code)
```

Comment:
This seems like a pretty exhaustive list of error codes, but in the case of an unexpected one, this will return `true`. Should this only return true for 200 status code?

Reply:
Hmm, you're right there. It would return `true` if it's an error. Those errors should raise an exception. 

What do you think about raising a `Fulfil::HttpError` whenever the status code indicates it's a non-OK status code (so, 400+) and it's not in the predefined list?

Reply:
I think that makes sense. That way nothing accidentally slips by if Fulfil starts responding with unexpected response codes

Reply:
What we also can do, is to always raise `Fulfil::HttpError` without looking at the specific HTTP status code. We would lose some specificity, but it makes things more simple and you would probably always rescue the `Fulfil::HttpError` anyway. 

On the other side, I do love the specificity as it's easier to recognize the different type of HTTP errors in services like Rollbar or Sentry.

Reply:
I'm in favor of the status code named classes